app/src/main/java/br/com/alexf/boraprofut/features/timer/TimerViewModel.kt

```diff
@@ -0,0 +1,60 @@
+package br.com.alexf.boraprofut.features.timer
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+data class TimerUiState(
+    val currentTime: Long = 0L,
+    val isPause: Boolean = true
+)
+
+class TimerViewModel(
+    private val timerUseCase: TimerUserCase
+) : ViewModel() {
+
+    private val _uiState = MutableStateFlow(TimerUiState())
+    val uiState = _uiState
+        .asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            timerUseCase.timer
+                .collectLatest { currentTime ->
+                    _uiState.update {
+                        it.copy(
+                            currentTime = currentTime,
+                            isPause = timerUseCase.isPause
+                        )
+                    }
+                }
+        }
+    }
+
+    fun setMinutes(minutes: Long) {
+        timerUseCase.timeMillis = minutes * 60000
+        _uiState.update {
+            it.copy(currentTime = timerUseCase.timeMillis)
+        }
+        timerUseCase.startTimer()
+    }
+
+    fun resume() {
+        timerUseCase.isPause = false
+        _uiState.update {
```

Comment:
Decidindo por seguir minha sugestão, adicionariamos a chamada de  timerUseCase.startTimer().
Essa seria a sugestão simples, poderia até ser melhorada talvez
```suggestion
        timerUseCase.startTimer()
        _uiState.update {
```

Reply:
Eu pensei em algo assim inicialmente, então pensei em como poderia fazer isso quando o `isPause` fosse modificado. Ai usei o `Delegates.observable` para não entrar em looping cada vez que `startTimer()` é executado